functions.php

```diff
@@ -54,5 +54,50 @@ function callPlant(array $plants): string
 }
 
 
+/**
+ * sanitizes and validates user input for new plant. Returns boolean of 'true' or 'false'
+ *
+ * @param $colloquial_name_input
+ * @param $latin_name_input
+ * @param $size_input
+ * @param $image_input
+ * @return boolean
+ */
+
+function validateNewPlant(string $colloquial_name_input, string $latin_name_input, string $size_input, string $image_input): bool
+{
+    $colloquial_name_input = filter_var($colloquial_name_input, FILTER_SANITIZE_STRING);
+    $latin_name_input = filter_var($latin_name_input, FILTER_SANITIZE_STRING);
+    $size_input = filter_var($size_input, FILTER_SANITIZE_STRING);
+    $image_input = filter_var($image_input, FILTER_SANITIZE_URL);
 
+    $colloquial_name_input = preg_match('/^[a-zA-Z\s]+$/', $colloquial_name_input);
+    $latin_name_input = preg_match('/^[a-zA-Z\s]+$/', $latin_name_input);
```

Comment:
this regex is allowing through name entries of space bars eg: ' '

Reply:
fixed